src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +69,20 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> workablePeople = new Predicate<Person>() {
+            @Override
+            public boolean test(Person person) {
+                if (person.getSex() == Person.Sex.WOMAN && person.getAge() >= fromAge
+                        && person.getAge() <= femaleToAge) {
+                    return true;
+                }
+                return (person.getSex() == Person.Sex.MAN && person.getAge() >= fromAge
+                        && person.getAge() <= maleToAge);
+            }
+        };
```

Comment:
I think it's better to use lambda here
```suggestion
        Predicate<Person> personPredicate = person -> person.getAge() >= fromAge
                && ((person.getSex() == Person.Sex.WOMAN && person.getAge() <= femaleToAge)
                || person.getSex() == Person.Sex.MAN && person.getAge() <= maleToAge);
```